gb-cpu/src/address_bus.rs

```diff
@@ -0,0 +1,221 @@
+pub mod operation;
+
+pub use operation::{FileOperation, RomOperation};
+
+/// AddressBus map specific range address to specific area like ROM/RAM.
+/// This Implementation of an AddressBus will be limited to 16-bit address
+pub struct AddressBus {
```

Comment:
From the documentation i'm reading at[Pan Docs](https://gbdev.io/pandocs/Memory_Map.html), I have several questions:

- Is this really the right place for the bios?
- Shouldn't we put in the not usable portion between FEA0 and FEFF?
- Shouldn't we split the ROM space in two parts to distinguish between the fixed and switchable parts of memory?

Reply:
1. For the bios position, read [Power-up sequence](https://gbdev.io/pandocs/Power_Up_Sequence.html),
   The Bios is the first portion of code to be executed on power-up so it's expected for the bios code to start at `0x0`
2. From the page you've linked:
  >  | FEA0 | FEFF | Not Usable | Nintendo says use of this area is prohibited |
  >  | -- | -- | -- | -- |
  One more thing, the area `0xfea0-0xfeff` is `95` long, not long enough for the bios it require `255`
3. `RomOnly` cartridge don't have swappable rom space